src/Microsoft.TemplateEngine.Edge/Settings/TemplateCache.cs

```diff
@@ -5,20 +5,28 @@
 using System.Collections.Generic;
 using System.Globalization;
 using System.Linq;
+using System.Text;
 using Microsoft.Extensions.Logging;
 using Microsoft.TemplateEngine.Abstractions;
+using Microsoft.TemplateEngine.Abstractions.TemplatePackage;
 using Newtonsoft.Json;
 using Newtonsoft.Json.Linq;
 
 namespace Microsoft.TemplateEngine.Edge.Settings
 {
     internal class TemplateCache
     {
+        private const string BulletSymbol = "\u2022";
+
         private readonly ILogger _logger;
 
-        public TemplateCache(ScanResult?[] scanResults, Dictionary<string, DateTime> mountPoints, ILogger logger)
+        public TemplateCache(IReadOnlyList<ITemplatePackage> allTemplatePackages, ScanResult?[] scanResults, Dictionary<string, DateTime> mountPoints, ILogger logger)
         {
             _logger = logger;
+
+            // this dictionary contains information about managed templates with overlapping identities
+            var overlappingIdentitiesMap = new Dictionary<DuplicatedIdentity, IList<(string TemplateName, string PackageId)>>(new DuplicatedIdentityComparer());
```

Comment:
would it be easier as 
```suggestion
            var overlappingIdentitiesMap = new Dictionary<string, IList<(ITemplateInfo template, string PackageId)>>();
```
where the key is identity.

then likely replacement logic used in CheckForOverlappingIdentity won't be needed, it would be enough to add the data.

Reply:
I need to have a path for both entries - the one from the key is used in the warning message too
![image](https://user-images.githubusercontent.com/95473390/221778848-7e544efe-fc3c-4c8c-937c-5897a589877f.png)